tests/lib/tools/fs-state

```diff
@@ -77,6 +80,66 @@ restore_file() {
     fi
 }
 
+start_monitor() {
+    if [ -n "$(command -v inotifywait)" ]; then
+        # In systems like arch-linux the inotifywait command fails when the output file does not exist
+        touch /tmp/fs.output
+        touch /tmp/fs.exclude
+
+        # Monitor files and directories creation
+        SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+
+        # $PWD is excluded because the backup dir is automatically restored
+        # .snap files created in tests/lib/snaps are excluded as they are not deleted to be reused
+        inotifywait -d -m -r -e CREATE -o /tmp/fs.output --exclude "($PWD|$PROJECT_PATH/tests/lib/snaps/.*/.*.snap)" /root /var/lib/snapd /var/snap /home "$SNAP_MOUNT_DIR"
+        retry -n 10 -wait 1 sh -c 'pgrep inotifywait &>/dev/null'  
+    fi
+}
+
+check_monitor() {
+    if [ -n "$(command -v inotifywait)" ]; then
+        # Stop inotifywait (it could be already stopped)
+        if pgrep inotifywait &>/dev/null; then
+            pkill inotifywait
+        fi
+        set +x
+        local created_files created_dirs missing
+        # the output file contains lines:
+        # /etc/ CREATE foo          # created file /etc/foo
+        # /etc/ CREATE,ISDIR foo    # created directory /etc/foo
+        created_files=$( sudo cat /tmp/fs.output | grep " CREATE " | awk '{ print $1$3 }' )
+        created_dirs=$( sudo cat /tmp/fs.output | grep " CREATE,ISDIR " | awk '{ print $1$3 }' )
+        missing=false
+
+        # Print the files that were created but not deleted
+        for file in $created_files; do
+            if [ -f "$file" ] || not grep -qx "$file" /tmp/fs.exclude; then
```

Comment:
I'm a bit confused by this (probably because I don't know what is the expected behaviour of `exclude_monitor`): here it looks like you are considering it an error if a file which is not in the exclusion list has been created, regardless of whether it has been deleted or not. Shouldn't the error be emitted only if the created file has not been deleted? In other words, shouldn't the `||` be a `&&`?

Reply:
This approach has change, the new one is using tree and diff tools. Please take a look.